rest/src/main/groovy/whelk/rest/api/Crud.groovy

```diff
@@ -241,7 +241,8 @@ class Crud extends HttpServlet {
                 doc.getCompleteId(), request.shouldEmbellish(), request.shouldFrame(), request.getLens())
 
         if (request.shouldEmbellish()) {
-            doc = whelk.loadEmbellished(doc.getShortId())
+            doc = doc.clone() // FIXME: this is because ETag calculation is done on non-embellished doc...
```

Comment:
Does this change API behaviour affecting the client now?

Reply:
Nope.
`loadEmbellished` had to be used before because the whole embellished doc was lazy-cached in `lddb__embellished`. But because of this the ETag calculation elsewhere in CRUD GET dependes on the doc object not changing.

(it might be better to change Whelk.embellish to return a new Document instance (and make Document immutable...))

Reply:
I see. Good catch! Indeed, the mutability is troublesome (mainly since we have no contract on it).